src/peer_select.cc

```diff
@@ -494,49 +578,77 @@ PeerSelector::selectMore()
         debugs(44, 3, "direct = " << DirectStr[direct]);
     }
 
-    if (!entry || entry->ping_status == PING_NONE)
-        selectPinned();
-    if (entry == NULL) {
-        (void) 0;
-    } else if (entry->ping_status == PING_NONE) {
-        selectSomeNeighbor();
+    if (selectionState == DoPreselection) {
+        if (!entry || entry->ping_status == PING_NONE)
```

Comment:
I suggest preserving wrong formatting for now, to expose the true code differences.